forge-gui/res/cardsfolder/k/kaylas_music_box.txt

```diff
@@ -3,7 +3,7 @@ ManaCost:2
 Types:Legendary Artifact
 A:AB$ Dig | Cost$ W T | DigNum$ 1 | ForceRevealToController$ True | ChangeNum$ All | DestinationZone$ Exile | RememberChanged$ True | ExileFaceDown$ True | SpellDescription$ Look at the top card of your library, then exile it face down. (You may look at it any time.) 
 S:Mode$ Continuous | MayLookAt$ You | Affected$ Card.ExiledWithSource+YouOwn | AffectedZone$ Exile
-A:AB$ Effect | Cost$ T | StaticAbilities$ MayPlay | SpellDescription$ Until end of turn, you may play cards you own exiled with CARDNAME.
+A:AB$ Effect | Cost$ T | StaticAbilities$ MayPlay | RememberObjects$ Remembered | SpellDescription$ Until end of turn, you may play cards you own exiled with CARDNAME.
```

Comment:
@Hanmac 
imo the affected cards shouldn't be determined just once

Like if you untap it after to exile another?

Reply:
> It looks at any cards you have currently exiled by the Music Box. It does not matter if they were exiled before or after the last activated ability has resolved.

so no need to remember?

Reply:
In theory, but then when it reenters it would still find the old ones

Reply:
how about checking with Defined/Present in the new StaticAbility if EffectSource is still with Card.StictlySelf on the battlefield? or something?

Reply:
I think once the effect is created it can use LKI if needed to decide exiled with?

Otherwise abilities like
> Sacrifice Bag of Holding: Return all cards exiled with Bag of Holding to their owner’s hand.

wouldn't work anyway?

Reply:
we need to separate between the ones that was before this left the battlefield
and the ones that have it returned

maybe we can compare timestamps for now?

Reply:
Can we try to move ExiledWith property into a CardCollection for the source instead of having it on each exiled one?

Imo this might help us to avoid additional remembering and
- finally get _Watcher for Tomorrow_ fixable
- allow _Prowl, Stoic Strategist_ to work cleanly (https://discord.com/channels/267367946135928833/550669370234372107/1034606225301721119)